File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
       
        int left = 0; 
        int right = prices.length -1; 
         int maxProfit =  0;
        int minPrice = 10000; 
        
        for(int price : prices) {
            minPrice = Math.min(minPrice, price); 
            maxProfit = Math.max(maxProfit, price - minPrice);
            
        }
        
        return maxProfit; 
        
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        left = 0  # Buy
        right = 1  # Sell
        max_profit = 0

        while right < len(prices):
            if prices[left] < prices[right]:
                current_profit = prices[right] - prices[left]
                max_profit = max(max_profit, current_profit)
            else:
                left = right
            right += 1

        return max_profit